src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+const successEl = (message) =>
+  `<div data-qa="notification" class="success">${message}</div>`;
+const errorEl = (message) =>
+  `<div data-qa="notification" class="error">${message}</div>`;
+
+const displayMessage = (message) => {
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('beforeend', message);
+};
+
+const promise1 = Promise.race([
+  new Promise((resolve) => {
+    const handleClick = () => {
+      resolve(successEl('First promise was resolved'));
+      document.removeEventListener('click', handleClick);
+    };
+
+    document.addEventListener('click', handleClick);
+  }),
+  new Promise((resolve, reject) => {
+    setTimeout(() => {
+      reject(errorEl('First promise was rejected'));
+    }, 3000);
+  }),
+]);
+
+const promise2 = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve(successEl('Second promise was resolved'));
+      document.removeEventListener('click', handleClick);
+      document.removeEventListener('contextmenu', handleClick);
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
+});
+
+let leftClick = false;
+let rightClick = false;
+
+const promise3 = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    } else if (e.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve(successEl('Third promise was resolved'));
+      document.removeEventListener('click', handleClick);
+      document.removeEventListener('contextmenu', handleClick);
+    }
```

Comment:
You are using two separate boolean variables to track the state of left and right clicks. This could be simplified by using a single variable that counts the number of unique clicks. This would also make the code more flexible if you need to handle more than two unique clicks in the future.